pkg/app/piped/appconfigreporter/appconfigreporter.go

```diff
@@ -207,15 +207,45 @@ func (r *Reporter) updateRegisteredApps(ctx context.Context, registeredAppPaths
 }
 
 // findRegisteredApps finds out registered application info in the given git repository.
-func (r *Reporter) findRegisteredApps(ctx context.Context, repoID string, repo gitRepo, lastScannedCommit, headCommitHash string, registeredAppPaths map[string]struct{}) ([]*model.ApplicationInfo, error) {
+func (r *Reporter) findRegisteredApps(ctx context.Context, repoID string, repo gitRepo, lastScannedCommit, headCommitHash string, registeredAppPaths map[string]string) ([]*model.ApplicationInfo, error) {
 	if lastScannedCommit == "" {
-		return r.scanAllFiles(repo.GetPath(), repoID, func(fileRelPath string) bool {
-			gitPathKey := makeGitPathKey(repoID, fileRelPath)
-			if _, registered := registeredAppPaths[gitPathKey]; !registered {
-				return true
+		// Scan all files because it seems to be the first commit since Piped starts.
+		apps := make([]*model.ApplicationInfo, 0)
+		err := fs.WalkDir(r.fileSystem, repo.GetPath(), func(path string, d fs.DirEntry, err error) error {
+			if err != nil {
+				return err
```

Comment:
Just want to ask, do we have any reason to not use that `scanAllFiles` function?

Reply:
@nghialv I feel you. Actually we can use the original `scanAllFiles()` removed just now. But we have to retrieve `appID` from `registeredAppPaths` only if called by `findRegisteredApps`. It is not a shared helper function anymore. It was not too huge function and originally it has been received `shouldSkip` which makes us confused.
So I settled on writing a bit overlapping logic directly into each function.